crates/fluvio/src/producer/partitioning.rs

```diff
@@ -48,28 +48,27 @@ impl Partitioner for SiphashRoundRobinPartitioner {
         config: &PartitionerConfig,
         maybe_key: Option<&[u8]>,
         _value: &[u8],
-    ) -> i32 {
+    ) -> PartitionId {
         match maybe_key {
             Some(key) => partition_siphash(key, config.partition_count),
             None => {
                 // Atomic increment. This will wrap on overflow, which is fine
                 // because we are only interested in the modulus anyway
                 let partition = self.index.fetch_add(1, Ordering::Relaxed);
-                partition % config.partition_count
+                partition % (config.partition_count as PartitionId)
```

Comment:
I think casting `PartitionCount` to `PartitionId` is not necessary here

Reply:
In that case, it is better to add comments around. 
In my opinion, the variant `partition mod count` is more readable than `partition mod (count as id)`